insights/config/specs.py

```diff
@@ -391,7 +391,7 @@
     "teamdctl_state_dump"       : CommandSpec("/usr/bin/teamdctl {iface} state dump", iface=r"team\S+"),
     "tomcat_web.xml"            : First([PatternSpec(r"etc/tomcat.*/web\.xml"),
                                     PatternSpec(r"conf/tomcat/tomcat.*/web\.xml")]),
-    "tomcat_virtual_dir_context": CommandSpec("/bin/grep -R --include '*.xml' 'VirtualDirContext' /usr/share/tomcat*"),
+    "tomcat_virtual_dir_context": CommandSpec("/usr/bin/find /usr/share -maxdepth 1 -name 'tomcat*' -exec grep -R -s 'VirtualDirContext' --include '*.xml' '{}' +"),
```

Comment:
Please correct me if something goes wrong here.
Since the `catalina.base` (or `catalina.home`) can be specified manually by the CU manually, how about using below pre-command to get the directory accurately?

~~~
pre_command:
catalina_home: "/bin/ps auxwww | grep java | grep '\-Dcatalina.base' | awk -F'-Dcatalina.base=' '{print $2}' | awk '{print $1}' | sort -u"
===

"tomcat_virtual_dir_context": CommandSpec("/bin/grep -R /bin/grep -R --include '*.xml' 'VirtualDirContext'  {catalina_home}/*", catalina_home=r"\S+"),
~~~

Reply:
That is a great idea. I think I will use this spec as a fallback for instances where tomcat is not running at the moment, but may want to have another one with a similar pre-command. Just note that `{catalina_home}/*` will not work as shell expansions are  disabled.

Reply:
@xiangce also, can pre-commands do pipes? I am asking because CommandSpecs cannot.

Reply:
I'm sure the pipe works in pre-commands (but doesn't in CommandSpecs.) And in the `insights/specs.py` of the master branch, you can implement it more simply :)